core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -704,12 +704,31 @@ private[spark] class DAGScheduler(
     assert(partitions.nonEmpty)
     val func2 = func.asInstanceOf[(TaskContext, Iterator[_]) => _]
     val waiter = new JobWaiter[U](this, jobId, partitions.size, resultHandler)
+    eagerPartitions(rdd)
     eventProcessLoop.post(JobSubmitted(
       jobId, rdd, func2, partitions.toArray, callSite, waiter,
       SerializationUtils.clone(properties)))
     waiter
   }
 
+  /**
+   * Responsible for eager evaluation of all dependency partitions.
+   * Takes effect only if <b>spark.rdd.eager.partitions</b> is true
+   * @param rdd : initial rdd to be evaluated
+   */
+  def eagerPartitions(rdd: RDD[_]): Unit = {
+    if (sc.getConf.get(config.EAGER_RDD_PARTITION_EVALUATE)) {
+      try {
+        rdd.dependencies.foreach { dependency =>
```

Comment:
doesn't look like you're ever evaluatiing `rdd.partitions` here.  I think that happens to be OK, because its evaluated in the calling functions before this ... but nonetheless I think it would be easier to reason about this if instead this called `rdd.partitions` and then just recursed on `rdd.dependencies`  `rdd.dependendies.foreach { dep => eagerPartitions(dep.rdd) }`

Reply:
i checked that all the caller methods were already doing ``rdd.partitions`` hence i skipped calling again, but yes, agree with your suggestion, will change ``eagerPartitions`` to call ``rdd.partitions`` for completeness